lab11.py

```diff
@@ -0,0 +1,80 @@
+from contextlib import contextmanager
+from threading import Lock, Thread
+DATA = {'old_item_0': 0, 'old_item_1': 1, 'old_item_2': 2, 'old_item_3': 3}
+
+
+class RWLock:
+    def __init__(self):
+        self.write_lock = Lock()
+        self.read_lock = Lock()
+        self.readers = 0
+
+    def acquire_read(self):
+        self.read_lock.acquire()
+        try:
+            self.readers += 1
+            # як тільки з'являєтсья перший читач, блокуємо write lock
+            if self.readers == 1:
+                self.write_lock.acquire()
+        finally:
+            self.read_lock.release()
```

Comment:
the algorithm here does not solve anything. None of the reading threads should be blocked unless there is a writer thread writing. In your approach writer thread will be deadlocked by readers

Reply:
but, here I thought, if someone read, anyone else cann't write in the same moment, if reader come earlier then writer. Is it bad?

Reply:
Also, if I understand correct, I don't need read_lock at all?

Reply:
please review the existing implementation of RW locks in other languages